packages/js/product-editor/src/utils/index.ts

```diff
@@ -44,6 +46,8 @@ export {
 	getProductTitle,
 	getProductVariationTitle,
 	getTruncatedProductVariationTitle,
+	handleConfirm,
+	handlePrompt,
```

Comment:
We have recently started being more strict about first introducing new exports as `__experimental` to give things time to prove themselves and settle before we commit to supporting backwards compatibility.

As such, let's export these as `__experimental`...

```js
handleConfirm as __experimentalHandleConfirm,
handlePrompt as __experimentalHandlePrompt,
```

Reply:
I *really* don't want to do that, and think that it that's the current practice it's really unhelpful (from a 3rd party developer point of view).

We don't want to release code that relies on 'experimental' APIs that are going to need updated. In this case we're giving ourselves future maintenance work to do even if the actual API is unchanged to update when the export changes to a non-prefixed version, and also introducing the risk of things breaking when that happens (is there an equivalent deprecation framework on the JS side to handle such transitions?).

Making this experimental just means that we won't use them, so we'd end up either implementing our own code instead, or just not integrating at all which doesn't help anyone.

Reply:
I can understand the hesitation to release code that relies on experimental APIs. That's precisely why we want to clearly label experimental APIs, so developers know that they are using something that may change/go away.

When things are moved from experimental to non-experimental, we will keep the `__experimental` prefixed version as well, so that existing code continues to work.

In this specific case, my main concern is that `handleConfirm` and `handlePrompt` both currently are implemented with the browser native dialogs. This is not ideal from a UX perspective, as those have many limitations. But, after thinking about it more, I think in this specific case the replacement API (using custom modals) would likely differ enough that we would create new methods anyway. And, these older methods could still co-exist with a new solution.

So, I think in this case we are probably okay without `__experimental` prefixes.

Reply:
Methods left named as-is and changelog entry added. Thanks